hpx/lcos/future.hpp

```diff
@@ -1127,18 +1164,21 @@ namespace hpx { namespace lcos
     // existing conversion path U --> T.
     template <typename T, typename U,
     HPX_CONCEPT_REQUIRES_(
-        boost::is_convertible<U, T>::value)>
-    hpx::future<T> make_future(hpx::future<U> && f)
+       !std::is_same<T, U>::value &&
+        std::is_convertible<U, T>::value)>
+    hpx::future<T> make_future(hpx::shared_future<U> f)
```

Comment:
Is this really something we want to have? the returned future isn't really unique, is it?